### PR TITLE
Cxr severity

### PR DESCRIPTION
**Pull request by mayaliliya**
# Pull Request Template
Added COVIDNet CXR-S model and COVIDxSev dataset for airspace severity grading. Built it off of pre-existing scripts and tested locally and was able to replicate results and run the default CXR-2 models as well.

Please add options that are relevant and mark any boxes that apply.

- [ ] Software (software that runs on the PC)
- [ ] Library (library that runs on the PC)
- [ ] Tool (tool that assists coding development)
- [ ] Other

## Type of change

Please mark any boxes that apply.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Checklist:

Please mark any boxes that have been completed.

- [X ] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X ] I have made corresponding changes to the documentation.
- [ X] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: docs/covidnet_severity.md [hunk 1]
```diff
@@ -1,3 +1,73 @@
...
+* logit tensor: `norm_dense_2/MatMul:0`
+* output tensor: `norm_dense_2/Softmax:0`
+* label tensor: `norm_dense_1_target:0`
+* class weights tensor: `norm_dense_1_sample_weights:0`
+* loss tensor: `Mean:0`
+
+### Steps for training
+To train the model the COVIDxSev dataset is required, to create the dataset please run [create_COVIDxSev.ipynb](../create_COVIDxSev.ipynb).
+TF training script from a pretrained model:
+1. We provide you with the tensorflow evaluation script, [train_tf.py](../train_tf.py)
```
Comment (haydengunraj) [inline docs/covidnet_severity.md:17]: Should be "training script"

File: docs/covidnet_severity.md [hunk 2]
```diff
@@ -1,3 +1,73 @@
...
+2. Locate the tensorflow checkpoint files (location of pretrained model)
+3. To train from the COVIDNet-CXR-S pretrained model:
+```
+python3 train_tf.py \
```
Comment (haydengunraj) [inline docs/covidnet_severity.md:21]: If you make any other changes, maybe change this to just "python" to match the rest of the docs

File: data.py [hunk 3]
```diff
@@ -212,4 +214,4 @@ def __getitem__(self, idx):
         class_weights = self.class_weights
         weights = np.take(class_weights, batch_y.astype('int64'))
 
-        return batch_x, keras.utils.to_categorical(batch_y, num_classes=self.n_classes), weights
+        return batch_x, keras.utils.to_categorical(batch_y, num_classes=self.n_classes), weights
```
Comment (haydengunraj) [inline data.py:217]: It would be best to roll back the removal of these newlines - PEP standard is to always have newlines, and this just adds another line change for no particular reason.

File: train_tf.py [hunk 4]
```diff
@@ -149,4 +160,4 @@
             print('Saving checkpoint at epoch {}'.format(epoch + 1))
 
 
-print("Optimization Finished!")
+print("Optimization Finished!")
```
Comment (haydengunraj) [inline train_tf.py:163]: See comment above about PEP standard and newlines.

File: docs/COVIDx.md [hunk 5]
```diff
@@ -1,4 +1,5 @@
 # COVIDx Dataset
+**Update 04/16/2021:Released COVIDxSev, a new airspace severity grading dataset for COVID-19 positive patients for COVIDNet CXR-S model.**\
```
Comment (haydengunraj) [inline docs/COVIDx.md:2]: Up to you whether you want to update the dates to match the actual merge date, which I'm guessing will be 04/18/2021. If you do end up changing them, consider also adding a space after the colon.

File: README.md [hunk 6]
```diff
@@ -4,6 +4,7 @@
 
 **Recording to webinar on [How we built COVID-Net in 7 days with Gensynth](https://darwinai.news/fny)**
 
+**Update 04/16/2021:** We released a new COVIDNet CXR-S [model](docs/models.md) and [COVIDxSev](create_COVIDxSev.ipynb) dataset for airspace severity grading in COVID-19 positive patient CXR images. For more information on training, testing and inference please refer to severity [docs](docs/covidnet_severity.md).
```
Comment (haydengunraj) [inline README.md:7]: Just marking this re: dates, see my comment on docs/COVIDx.md for context.

**Comment by haydengunraj** [comment]
On my local machine, I'm able to construct the dataset and run training and eval without any issues. My numbers match for both the dataset sizes and accuracy.

I added a few comments for minor nitpicky things, but overall everything looks good.

**Review by andyzzhao — COMMENTED** [review]
All changes are working on my end as well. Nice work Maya!

**Review by haydengunraj — APPROVED** [review]
Approving now, but have a look at the comments and make any desired changes before merging.
